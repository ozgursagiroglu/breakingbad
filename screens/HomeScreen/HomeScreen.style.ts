import {Dimensions, StyleSheet, TextStyle, ViewStyle} from 'react-native';

type Style = {
  root: ViewStyle;
  title: TextStyle;
  row: ViewStyle;
  item: ViewStyle;
  layer: ViewStyle;
  name: TextStyle;
};

const SCREEN_PADDING = 16;
const GRID_GAP = 16;
// (screenWidth - ((screenPadding * 2) + (grid gap))) / 2
const ITEM_WIDTH =
  (Dimensions.get('screen').width - (SCREEN_PADDING * 2 + GRID_GAP)) / 2;

export default StyleSheet.create<Style>({
  root: {
    flex: 1,
  },
  title: {
    fontWeight: '600',
    fontSize: 24,
    textAlign: 'center',
    paddingVertical: 24,
  },

  row: {
    paddingHorizontal: 16,
    justifyContent: 'space-between',
  },

  item: {
    position: 'relative',
    width: ITEM_WIDTH,
    height: ITEM_WIDTH * 1.5,
    marginBottom: 16,
  },

  layer: {
    paddingHorizontal: 16,
    paddingVertical: 8,

    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
  },

  name: {
    fontSize: 16,
    color: 'white',
  },
});
