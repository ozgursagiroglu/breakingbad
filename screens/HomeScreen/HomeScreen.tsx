import React from 'react';
import {Text, View} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import CharacterList from '../../components/CharacterList/CharacterList';
import AppLayout from '../../components/common/AppLayout/AppLayout';
import styles from './HomeScreen.style';

type Props = {};

const HomeScreen: React.FC<Props> = ({}) => {
  const insets = useSafeAreaInsets();

  return (
    <AppLayout>
      <View style={{paddingTop: insets.top}}>
        <Text style={styles.title}>Breaking Bad Characters</Text>
      </View>
      <CharacterList />
    </AppLayout>
  );
};

export default HomeScreen;
