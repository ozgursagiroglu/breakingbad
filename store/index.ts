import {combineReducers, configureStore} from '@reduxjs/toolkit';
import {setupListeners} from '@reduxjs/toolkit/dist/query';
import {characterService} from './services/characterService';

export const store = configureStore({
  reducer: combineReducers({
    [characterService.reducerPath]: characterService.reducer,
  }),
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware().concat([characterService.middleware]),
});

setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
