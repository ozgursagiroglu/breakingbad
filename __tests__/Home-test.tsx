import React from 'react';
import {render} from '@testing-library/react-native';
import HomeScreen from '../screens/HomeScreen/HomeScreen';
import {Provider} from 'react-redux';
import store from '../store';
import Navigation from '../navigation';
import {act} from 'react-test-renderer';

const markup = (
  <Provider store={store}>
    <Navigation>
      <HomeScreen />
    </Navigation>
  </Provider>
);

describe('HomeScreen', () => {
  it('renders correctly', async () => {
    const {toJSON} = render(markup);

    await act(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('title was render correctly', () => {
    const {getByText} = render(markup);

    expect(getByText('Breaking Bad Characters')).toBeTruthy();
  });
});
