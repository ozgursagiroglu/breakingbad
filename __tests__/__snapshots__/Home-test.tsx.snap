// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#f5f5f5",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingTop": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 24,
                  "fontWeight": "600",
                  "paddingVertical": 24,
                  "textAlign": "center",
                }
              }
            >
              Breaking Bad Characters
            </Text>
          </View>
          <RCTScrollView
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            testID="items"
            viewabilityConfigCallbackPairs={Array []}
          >
            <View />
          </RCTScrollView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
